crates/bdk/src/wallet/mod.rs

```diff
@@ -224,6 +227,24 @@ impl Wallet {
             NewError::Persist(_) => unreachable!("no persistence so it can't fail"),
         })
     }
+
+    /// Infallibly return a derived address using the external descriptor, see [`AddressIndex`] for
+    /// available address index selection strategies. If none of the keys in the descriptor are derivable
+    /// (i.e. does not end with /*) then the same address will always be returned for any [`AddressIndex`].
+    pub fn get_address(&mut self, address_index: AddressIndex) -> AddressInfo {
+        self.try_get_address(address_index).unwrap()
+    }
+
+    /// Infallibly return a derived address using the internal (change) descriptor.
```

Comment:
I'm curious to know why we needed to create infallible version of the address functions.

Reply:
I think the only reason for this is so that non-persistent / memory only applications don't need to handle an persistence error which will never happen. This is probably a rare case but I suspect there are some embedded system applications this would be useful.

Reply:
This is an excellent addition, thank you @notmandatory.